src/ACEtk/interpretation/ContinuousEnergyNeutron/CrossSection/src/ctor.hpp

```diff
@@ -0,0 +1,15 @@
+CrossSection( Slice energies, Slice values ) :
+  energyGrid_( energies ),
+  values_( values )
+{ 
+  if( energyGrid_.size() != values_.size() ){
+    Log::error( "Energy grid size must equal size of cross section values." );
+    throw std::exception();
+  }
+}
+
+template< typename Range  >
```

Comment:
There's no reason the `energies` and `values` need be the same type. I would prefer to also constrain this parameter to the required interface using SFINAE